src/include/packing_inline.h

```diff
@@ -231,19 +231,15 @@ __pack_next(WT_PACK *pack, WT_PACK_VALUE *pv)
         case 'b':                                                  \
         case 'h':                                                  \
         case 'i':                                                  \
+        case 'l':                                                  \
             (pv).u.i = va_arg(ap, int);                            \
             break;                                                 \
         case 'B':                                                  \
         case 'H':                                                  \
         case 'I':                                                  \
         case 't':                                                  \
-            (pv).u.u = va_arg(ap, unsigned int);                   \
-            break;                                                 \
-        case 'l':                                                  \
-            (pv).u.i = va_arg(ap, long);                           \
-            break;                                                 \
         case 'L':                                                  \
-            (pv).u.u = va_arg(ap, unsigned long);                  \
+            (pv).u.u = va_arg(ap, unsigned int);                   \
```

Comment:
How about a comment here, and in the b/h/i/l case above.  That int must be used because 1) compilers promote to smaller sizes to int for varargs arguments. 2) that 'l' accommodates a 4 byte integer.